backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/provider/NodeLabelPositionProvider.java

```diff
@@ -41,12 +170,22 @@ public Position getPosition(NodeLayoutData node, LabelLayoutData label) {
             y = -(label.getTextBounds().getSize().getHeight() + LayoutOptionValues.DEFAULT_ELK_NODE_LABELS_PADDING);
             break;
         default:
-            x = (node.getSize().getWidth() - label.getTextBounds().getSize().getWidth()) / 2;
-            y = 0;
             break;
         }
-
-        return Position.at(x, y);
+        return y;
     }
 
+    private boolean isOutside(EnumSet<NodeLabelPlacement> nodeLabelPlacementSet) {
+        boolean outside = false;
+        for (NodeLabelPlacement nodeLabelPlacement : nodeLabelPlacementSet) {
+            switch (nodeLabelPlacement) {
+            case OUTSIDE:
+                outside = true;
+                break;
+            default:
+                break;
+            }
+        }
```

Comment:
There are a lot of `for` / `switch` in this PR which does not make the code very easy to read. Here this code could easily be replaced by a `nodeLabelPlacementSet.stream().anyMatch(NodeLabelPlacement.OUTSIDE::equals)`.

Reply:
Done